apps/condo/domains/property/components/BasePropertyForm/index.tsx

```diff
@@ -125,15 +128,43 @@ const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
                                     </Form.Item>
                                 </Col>
                                 <Col span={24} >
-                                    <Form.Item name='map' hidden >
+                                    <Form.Item
+                                        hidden
+                                        name='map'
+                                        rules={[
+                                            {
+                                                validator (rule, value) {
+                                                    const unitLabels = value?.sections
+                                                        ?.map((section) => section.floors
+                                                            ?.map(floor => floor.units
+                                                                ?.map(unit => unit.label)
+                                                            )
+                                                        )
+                                                        .flat(2)
+
+                                                    if (unitLabels && unitLabels.length !== new Set(unitLabels).size) {
+                                                        setMapValidationError(SameUnitNamesErrorMsg)
+                                                        return Promise.reject()
+                                                    }
+
+                                                    setMapValidationError(null)
+                                                    return Promise.resolve()
+                                                },
+                                            },
+                                        ]}
+                                    >
                                         <Input />
                                     </Form.Item>
-                                    <Form.Item shouldUpdate={true}>
+                                    <Form.Item
+                                        shouldUpdate={true}
+                                        onFocus={() => setMapValidationError(null)}
```

Comment:
Do we really need to validate the input on focus? I think it would better to validate this on blur. Wyt?